### DISCUSSION TITLE
TEM in pyiron

### DISCUSSION THREAD

**Discussion by jan-janssen** [pyiron/pyiron_experimental#122] [Team Posts]
Hi Ning, 

To speed things up I just want to send you two tutorials: 
https://github.com/pyiron/pyiron/blob/master/notebooks/first_steps.ipynb
https://github.com/pyiron/pyiron/blob/master/notebooks/data_mining.ipynb

These can be executed online, using mybinder: 
https://mybinder.org/v2/gh/pyiron/pyiron/master

Still I recommend to install pyiron locally to already get setup for the development process: 
https://pyiron.readthedocs.io/en/latest/source/installation.html#installlocal

For implementing your own classes in pyiron, I have two examples one for python codes like TEMMETA which Niels developed: 
https://github.com/jan-janssen/temmeta-example

And finally one for calling external executables based on shell scripts: 
https://github.com/jan-janssen/match-series-example/blob/master/pyiron.ipynb

Based on these tutorials you should be able to implement pyiron wrappers for your classes. I would recommend that you present those in the pyiron meeting - maybe already next week friday 14:00, then I can help you with the configuration on the cluster. I assume you are using the TALOS cluster, so we would have to configure the queuing system to work with pyiron. 

We have example configurations available at:
https://github.com/pyiron/pysqa/tree/master/tests/config/slurm
The configuration of the queuing system can sometimes be a bit tricky so I would recommend to try it with something like the Lammps jobs first, just to validate that everything is working correctly. 

Inside the pyiron project we try to use Github for most of our communication, to have everything on one place. So I just resend you the invitation on Github to join the pyiron group. We can then create a subgroup for this project and have the rest of the discussion there. My goal of this collaboration would be to have some tutorial which helps people with implementing their own codes in pyiron. 

Best regards, 

Jan

**Comment by jan-janssen** [comment]
To structure things a little bit and take into account the emails from today, I see the following todos: 

- Create a conda forge package for @NingWang1990 code.
- Setup pyiron for @NingWang1990 . I am still not sure which cluster you are using. 
- Start by writing a small pyiron wrapper for @NingWang1990 code - mainly for educational purposes. 

Once this is done, the next step would be the integration of the TEM simulations, but for those we have to wait for the lecture from Christian Liebscher.

**Comment by jan-janssen** [comment]
To keep track of the different experimental activities: 

- https://github.com/din14970/NRR-emd - TEMMETA + match-series
- https://github.com/din14970/TEMMETA-segmentation-example - TEMMETA + pystem 
- https://github.com/jan-janssen/match-series-example - pyiron + match-series
- https://github.com/jan-janssen/temmeta-example - pyiron + TEMMETA

**Comment by jan-janssen** [comment]
Maybe some additional resources how to implement new classes https://github.com/pyiron/pyiron-meeting/tree/master/2020-05-15/jan-janssen

**Comment by jan-janssen** [comment]
Ning configured pyiron on TALOS the configuration is available on https://github.com/NingWang1990/pyiron_talos

**Comment by NingWang1990** [comment]
Hi, Jan, I uploaded a wrapper for pyprismatic into my github. https://github.com/NingWang1990/pyiron_pyprismatic Currently only several parameters can be passed from the wrapper to prismatic.  The simple test looks fine. 
The installation of pyprismatic in Talos is a little bit tricky, Yesterday I sent a request to garching people. They may install it for us.

**Comment by jan-janssen** [comment]
Hi Ning, 

I tested your example on mybinder and noticed you are using pyprismatic 1.1.16 from 2017:
https://pypi.org/project/PyPrismatic/#history
I am not sure what is the recommendation from Christian Liebscher, but there seems to be a new version 1.2.1 from july 2020 available on conda: 
https://anaconda.org/conda-forge/pyprismatic
The issue with the new version is they changed the output format to HDF5. There website states it should be possible to open these files with py4dstem: 
https://prism-em.com/docs-outputs/

I tried both in: 
https://github.com/jan-janssen/prismatic-example
The new version works fine, but opening the file with py4dstem does not work. So I use h5py to browse the file instead.

**Comment by NingWang1990** [comment]
Hi, Jan, 
Thanks a lot! there is no special reason to install version 1.1.16. I simply followed their installing instruction, which is apparently not updated in time.  I just told the conda version to the colleagues in Garching.  I think they will install the latest one into Talos. I think py4dstem is not very necessary, we may leave it out at the moment.
